mypyc/genops.py

```diff
@@ -2734,7 +2735,7 @@ def accept(self, node: Statement) -> None: ...
 
     def accept(self, node: Union[Statement, Expression]) -> Optional[Value]:
         try:
-            if isinstance(node, Expression):
+            if isinstance(node, Expression) and not isinstance(node, YieldExpr):
```

Comment:
Is this really needed? Why not just make `visit_yield_expr` return the type it is supposed to.

Reply:
Fixed.